### ISSUE TITLE
az aks update - to attach acr 

### ISSUE THREAD

**Issue by plooploops** [Azure/azure-cli-extensions#2517]
## Describe the bug

AKS w/User Assigned Managed Identity unable to attach ACR with az aks update -g $RG -n $CLUSTER_NAME --attach-acr {} command.

**Command Name**
`az aks update
Extension Name: aks-preview. Version: 0.4.62.`

**Errors:**
```
ValidationError: Could not create a role assignment for ACR. Are you an Owner on this subscription?
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

Create an AKS cluster (without yet attaching acr) with user assigned managed identity.

Create an Azure Container Registry in the same resource group.

Create a User Assigned Managed Identity and assign it to the RG with AKS (not the MC_ resource group).

USER_ASSIGNED_IDENTITY=$(az identity create -g $RG -n $USER_ASSIGNED_IDENTITY_NAME)

- `az aks update -g $RG -n $CLUSTER_NAME --attach-acr {}`

## Expected Behavior
Able to attach ACR to an AKS cluster.  The workaround is to attach ACR upon cluster creation (az aks create --attach-acr), or else to explicitly assign the user assigned managed identity the role 'AcrPull' with scope to the ACR Resource ID.

## Environment Summary
```
Linux-4.4.0-19041-Microsoft-x86_64-with-glibc2.29
Python 3.8.2
Installer: 

azure-cli 2.12.1 *

Extensions:
aks-preview 0.4.62
azure-firewall 0.6.0

```
## Additional Context

Attempted in wsl1:
![image](https://user-images.githubusercontent.com/2498998/95894102-7d393780-0d3d-11eb-8255-5d443073579a.png)

**Comment by plooploops** [comment]
Related: https://github.com/Azure/AKS/issues/1517

**Comment by yonzhan** [comment]
aks

**Comment by scgbear** [comment]
<details>
<summary>Internal Tracking</summary>
devdivcsef 396067
</details>

**Comment by precisely4ajeet** [comment]
Any workaround for this issue ? We are also facing same issue.

**Comment by rbavery** [comment]
I am also getting this problem, what's a simple workaround that doesn't require me to be an owner?

**Comment by CD-UNCC** [comment]
I am subscription owner and have this issue as well

**Comment by yonzhan** [comment]
ask service team should take a look

**Comment by miwithro** [comment]
@CD-UNCC what command did you run to add acr?  I just ran the below as an owner and it worked.

az aks update --resource-group <> --name <> --attach-acr <ResourceID>

We are working on a feature in Q3 that will eliminate the requirement to do the attach with Owner rights.

**Comment by CD-UNCC** [comment]
@miwithro The exact command you posted with the addition of the arc name. This issue occurred over night when our cluster suddenly could no longer pull new images. Yesterday someone at azure assigned our subscription to a new billing account, then this happened. I am still the owner and administrator of this subscription and have created all of these resources.

**Comment by miwithro** [comment]
so you tried with the acr name or the Resource ID of the acr?

**Comment by miwithro** [comment]
can you try with the ResourceID?

**Comment by CD-UNCC** [comment]
@miwithro exact same failure `Waiting for AAD role to propagate[################################    ]  90.0000%Could not create a role assignment for ACR. Are you an Owner on this subscription?`

**Comment by miwithro** [comment]
@CD-UNCC can you create a support case so we can have an Engineer look at what is wrong specific to your environment?

**Comment by miwithro** [comment]
@CD-UNCC were you able to create a support case?  Also in the interim can you upgrade your aks-preview version which I see is running 0.4.62.  On my test I am running aks-preview 0.5.19.

**Comment by CD-UNCC** [comment]
@miwithro What we came to find out was that in having our azure account migrated to a different billing subscription literally breaks kubernetes. We found that a few services simple cannot survive this while most can. The only solution was to create a new cluster after the migration. Everything is fine now. Luckily this was not production. I'm sure deep down some technical support could have solved this issue but time was not something we had much of.
